navigation_and_routing/lib/src/screens/author_details.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../data.dart';
+import '../widgets/book_list.dart';
+import 'book_details.dart';
+
+class AuthorDetailsScreen extends StatelessWidget {
+  final Author author;
+
+  const AuthorDetailsScreen({
+    Key? key,
+    required this.author,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(author.name),
+      ),
+      body: Center(
+        child: Column(
+          children: [
+            Expanded(
+              child: BookList(
+                books: author.books,
+                onTap: (book) {
+                  Navigator.of(context).push<dynamic>(
```

Comment:
Is it on purpose that this uses the page-less imperative API of the navigator? Why is it not navigating to `/books/:bookId`?